src/app/Agent/AgentProbeTemplates.tsx

```diff
@@ -0,0 +1,442 @@
+  );
+
+  const handleError = React.useCallback(
+    (error) => {
+      setIsLoading(false);
+      setErrorMessage(error.message);
+    },
+    [setIsLoading, setErrorMessage]
+  );
+
+  const refreshTemplates = React.useCallback(() => {
+    setIsLoading(true);
+    addSubscription(
+      context.target
+        .target()
+        .pipe(
+          concatMap((target) => context.api.getProbeTemplates()),
+          first()
+        )
+        .subscribe(
+          (value) => handleTemplates(value),
+          (err) => handleError(err)
+        )
+    );
+  }, [addSubscription, context, context.target, context.api, setIsLoading, handleTemplates, handleError]);
+
+  React.useEffect(() => {
+    addSubscription(
+      context.target.target().subscribe(() => {
+        setFilterText('');
+        refreshTemplates();
+      })
+    );
+  }, [context, context.target, addSubscription, refreshTemplates]);
+
+  React.useEffect(() => {
+    if (!context.settings.autoRefreshEnabled()) {
+      return;
+    }
+    const id = window.setInterval(
+      () => refreshTemplates(),
+      context.settings.autoRefreshPeriod() * context.settings.autoRefreshUnits()
+    );
+    return () => window.clearInterval(id);
+  }, []);
+
+  React.useEffect(() => {
+    const sub = context.target.authFailure().subscribe(() => {
+      setErrorMessage('Auth failure');
+    });
+    return () => sub.unsubscribe();
+  }, [context.target]);
+
+  const displayTemplates = React.useMemo(
+    () => filteredTemplates.map((t: ProbeTemplate) => [t.name, t.xml]),
+    [filteredTemplates]
+  );
+
+  const handleDelete = (rowData) => {
+    addSubscription(
+      context.api
+        .deleteCustomProbeTemplate(rowData[0])
+        .pipe(first())
+        .subscribe(() => {
+          refreshTemplates();
+        })
+    );
+  };
+
+  const handleInsert = (rowData) => {
+    addSubscription(
+      context.api
+        .insertProbes(rowData[0])
+        .pipe(first())
+        .subscribe(() => {})
+    );
+  };
+
+  const actionResolver = (rowData: IRowData, extraData: IExtraData) => {
+    if (typeof extraData.rowIndex == 'undefined') {
+      return [];
+    }
+    let actions = [
+      {
+        title: 'Insert Probes...',
+        onClick: (event, rowId, rowData) => handleInsert(rowData),
+      },
+    ] as IAction[];
+    actions = actions.concat([
+      {
+        isSeparator: true,
+      },
+      {
+        title: 'Delete',
+        onClick: (event, rowId, rowData) => handleDeleteButton(rowData),
+      },
+    ]);
+    return actions;
+  };
+
+  const handleModalToggle = () => {
+    setModalOpen((v) => {
+      if (v) {
+        setUploadFile(undefined);
+        setUploadFilename('');
+        setUploading(false);
+      }
+      return !v;
+    });
+  };
+
+  const handleFileChange = (value, filename) => {
+    setFileRejected(false);
+    setUploadFile(value);
+    setUploadFilename(filename);
+  };
+
+  const handleUploadSubmit = () => {
```

Comment:
Same suggestion from Max's comment https://github.com/cryostatio/cryostat-web/pull/558/files#r1002156730

Reply:
I think this should apply to all callbacks defined here.